src/macroxs_header.F90

```diff
@@ -0,0 +1,919 @@
+          norm = sum(temp_energy(:,gin,iazi,ipol))
+          if (norm > ZERO) then
+            temp_energy(:,gin,iazi,ipol) = temp_energy(:,gin,iazi,ipol) / norm
+          end if
+        end do
+
+        if (scatt_type == ANGLE_LEGENDRE .and. legendre_mu_points /= 1) then
+          call this % scatter(iazi, ipol) % obj % init(legendre_mu_points, &
+                 temp_energy(:,:,iazi,ipol), temp_mult(:,:,iazi,ipol), &
+                 scatt_coeffs(:,:,:,iazi,ipol))
+        else
+          call this % scatter(iazi, ipol) % obj % init(this % order, &
+                 temp_energy(:,:,iazi,ipol), temp_mult(:,:,iazi,ipol), &
+                 scatt_coeffs(:,:,:,iazi,ipol))
+        end if
+
+      end do
+    end do
+
+    ! Now go through and normalize chi
+    if (mat % fissionable) then
+      do ipol = 1, npol
+        do iazi = 1, nazi
+          do gin = 1, groups
+            ! Normalize Chi
+            norm =  sum(this % chi(:,gin,iazi,ipol))
+            if (norm > ZERO) then
+              this % chi(:,gin,iazi,ipol) = this % chi(:,gin,iazi,ipol) / norm
+            end if
+          end do
+        end do
+      end do
+    end if
+
+    ! Deallocate temporaries for the next material
+    deallocate(scatt_coeffs, temp_energy, temp_mult)
+
+  end subroutine macroxs_angle_init
+
+!===============================================================================
+! MACROXS*_CLEAR resets and deallocates data in MacroXS.
+!===============================================================================
+
+  subroutine macroxs_iso_clear(this)
+
+    class(MacroXS_Iso), intent(inout) :: this ! The MacroXS to clear
+
+    if (allocated(this % total)) then
+      deallocate(this % total, this % absorption, &
+                 this % nu_fission)
+    end if
+
+    if (allocated(this % fission)) then
+      deallocate(this % fission)
+    end if
+
+    if (allocated(this % k_fission)) then
+      deallocate(this % k_fission)
+    end if
+
+    call this % scatter % clear()
+
+    if (allocated(this % chi)) then
+      deallocate(this % chi)
+    end if
+
+  end subroutine macroxs_iso_clear
+
+  subroutine macroxs_angle_clear(this)
+
+    class(MacroXS_Angle), intent(inout) :: this ! The MacroXS to clear
+    integer :: i, j
+
+    if (allocated(this % total)) then
+      deallocate(this % total, this % absorption, &
+                 this % nu_fission)
+    end if
+
+    if (allocated(this % fission)) then
+      deallocate(this % fission)
+    end if
+
+    if (allocated(this % k_fission)) then
+      deallocate(this % k_fission)
+    end if
+
+    do i = 1, size(this % scatter,dim=2)
+      do j = 1, size(this % scatter,dim=1)
+        call this % scatter(j,i) % obj % clear()
+      end do
+    end do
+    if (allocated(this % scatter)) then
+      deallocate(this % scatter)
+    end if
+
+    if (allocated(this % chi)) then
+      deallocate(this % chi)
+    end if
+
+  end subroutine macroxs_angle_clear
```

Comment:
These `clear` subroutines are not needed since `allocatable` variables are deallocated by default.  Is that right, @paulromano?